maat-court-data-api/src/main/java/gov/uk/courtdata/unlink/service/UnlinkListener.java

```diff
@@ -27,13 +31,19 @@ public class UnlinkListener {
 
     private final QueueMessageLogService queueMessageLogService;
 
-
-    @JmsListener(destination = "${cloud-platform.aws.sqs.queue.unlink}")
-    public void receive(@Payload final String message) {
-
-        MDC.put(LoggingData.REQUEST_TYPE.getValue(), MessageType.UNLINK.name());
-        queueMessageLogService.createLog(MessageType.UNLINK, message);
-        Unlink unlink = gson.fromJson(message, Unlink.class);
-        unLinkProcessor.process(unlink);
+    @SqsListener(value = "${cloud-platform.aws.sqs.queue.unlink}",
+            deletionPolicy = SqsMessageDeletionPolicy.ON_SUCCESS)
+    public void receive(@Payload final String message,
+                        final @Headers MessageHeaders headers) {
+
+        try {
+            log.debug("message-id {}", headers.get("MessageId"));
+            MDC.put(LoggingData.REQUEST_TYPE.getValue(), MessageType.UNLINK.name());
+            queueMessageLogService.createLog(MessageType.UNLINK, message);
+            Unlink unlink = gson.fromJson(message, Unlink.class);
+            unLinkProcessor.process(unlink);
+        } catch (ValidationException exception) {
+            log.warn(exception.getMessage());
```

Comment:
What level are the production logs? For example, if the production logs are at ERROR level we would not log the failing message?

I think there is also a potential security consideration for logging messages in the logs FYI. For example, if a hacker gains entry to the logs could they reverse engineer the payload and obtain some data.

Reply:
This is an interesting point - the log level is only set to INFO at the moment.

Reply:
Ah wait we are receiving WARN and ERROR messages, ignore me!